contracts/0.8.25/vaults/StakingVault.sol

```diff
@@ -304,22 +321,26 @@ contract StakingVault is IStakingVault, IBeaconProxy, BeaconChainDepositLogistic
 
     /**
      * @notice Performs a deposit to the beacon chain deposit contract
-     * @param _numberOfDeposits Number of deposits to make
-     * @param _pubkeys Concatenated validator public keys
-     * @param _signatures Concatenated deposit data signatures
+     * @param _deposits Array of deposit structs
      * @dev Includes a check to ensure StakingVault is balanced before making deposits
      */
-    function depositToBeaconChain(
-        uint256 _numberOfDeposits,
-        bytes calldata _pubkeys,
-        bytes calldata _signatures
-    ) external {
-        if (_numberOfDeposits == 0) revert ZeroArgument("_numberOfDeposits");
-        if (!isBalanced()) revert Unbalanced();
+    function depositToBeaconChain(Deposit[] calldata _deposits) external {
+        if (_deposits.length == 0) revert ZeroArgument("_deposits");
         if (msg.sender != _getStorage().nodeOperator) revert NotAuthorized("depositToBeaconChain", msg.sender);
+        if (!isBalanced()) revert Unbalanced();
+
+        uint256 numberOfDeposits = _deposits.length;
+        for (uint256 i = 0; i < numberOfDeposits; i++) {
+            Deposit calldata deposit = _deposits[i];
+            BEACON_CHAIN_DEPOSIT_CONTRACT.deposit{value: deposit.amount}(
+                deposit.pubkey,
+                bytes.concat(withdrawalCredentials()),
+                deposit.signature,
+                deposit.depositDataRoot
+            );
+        }
 
-        _makeBeaconChainDeposits32ETH(_numberOfDeposits, bytes.concat(withdrawalCredentials()), _pubkeys, _signatures);
-        emit DepositedToBeaconChain(msg.sender, _numberOfDeposits, _numberOfDeposits * 32 ether);
+        emit DepositedToBeaconChain(msg.sender, numberOfDeposits);
```

Comment:
Wouldn't a total deposited amount be needed for analytics?

Reply:
I've decided not to include the total amount because each individual amount is included in the DepositContract's `DepositEvent`

Reply:
I'm not an expert, but wouldn't it be more convenient for analytics / UI to receive it as only one event instead of many? Just from this angle.